libqtile/core/session_manager.py

```diff
@@ -24,35 +34,43 @@ def __init__(
         :param state:
             The state to restart the qtile instance with.
         """
-        self.eventloop = asyncio.new_event_loop()
-        asyncio.set_event_loop(self.eventloop)
+        lifecycle.behavior = lifecycle.behavior.TERMINATE
 
-        self.qtile = Qtile(kore, config, self.eventloop, no_spawn=no_spawn, state=state)
+        self.qtile = Qtile(kore, config, no_spawn=no_spawn, state=state)
+        self.server = ipc.Server(
+            self._prepare_socket(fname),
+            self.qtile.server.call,
+        )
 
-        if fname is None:
+    def _prepare_socket(self, path: Optional[str] = None) -> str:
+        if path is None:
             # Dots might appear in the host part of the display name
             # during remote X sessions. Let's strip the host part first
-            display_name = kore.display_name
+            display_name = self.qtile.core.display_name
             display_number = display_name.partition(":")[2]
             if "." not in display_number:
                 display_name += ".0"
-            fname = ipc.find_sockfile(display_name)
+            path = ipc.find_sockfile(display_name)
```

Comment:
Again, I'm good with the renaming but please put it in another comment, it hurts the readability of the diff.